go/libraries/doltcore/merge/merge.go

```diff
@@ -296,7 +296,8 @@ func calcTableMergeStats(ctx context.Context, tbl *doltdb.Table, mergeTbl *doltd
 	return ms, nil
 }
 
-type rowMerger func(ctx context.Context, nbf *types.NomsBinFormat, sch schema.Schema, r, mergeRow, baseRow types.Value) (types.Value, bool, error)
+// returns merged roow, bool indicating if a cell-wise merge was performed, bool indicating a conflict, and error
+type rowMerger func(ctx context.Context, nbf *types.NomsBinFormat, sch schema.Schema, r, mergeRow, baseRow types.Value) (types.Value, bool, bool, error)
```

Comment:
4 return values is too many. You should either separate `rowMerger` logic into multiple steps or create a new type to hold the returned information